src/couch_replicator_manager.erl

```diff
@@ -338,16 +364,18 @@ terminate(_Reason, State) ->
 
 code_change(1, State, _Extra) ->
     {ok, erlang:append_element(State, [node() | nodes()])};
+code_change(2, State, _Extra) ->
+    {ok, erlang:append_element(State, make_ref())};
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
 
-start_changes_reader(DbName, Since) ->
-    spawn_link(?MODULE, changes_reader, [self(), DbName, Since]).
+start_changes_reader(DbName, Since, Epoch) ->
+    spawn_link(?MODULE, changes_reader, [{self(), Epoch},  DbName, Since]).
```

Comment:
Ah, you'd break upgrades here, so you might as well remove the code_change support.